extra/migrate_settings.rb

```diff
@@ -83,6 +94,18 @@ def migrate(data)
   return output, data
 end
 
+def migrate_dns_provider_name(aname)
+  if recognized_dns_provider_name?(aname)
```

Comment:
rolled back changes to migrate_settings and migration_test, as those are going to be replaced by the new settings migrations (kept dns_settings_migrations for now as a reference, will convert to settings migration once the PR has been merged)